apps/server/src/modules/room/repo/entity/room-content.entity.ts

```diff
@@ -0,0 +1,23 @@
+import { Embeddable, Embedded, Entity, Property } from '@mikro-orm/core';
+import { RoomContentType } from '../../domain/type';
+import { BaseEntityWithTimestamps } from '@shared/domain/entity';
+import { EntityId } from '@shared/domain/types';
+import { ObjectIdType } from '@shared/repo/types/object-id.type';
+
+@Embeddable()
+class RoomContentItem {
+	@Property({ nullable: false })
+	id!: EntityId;
+
+	@Property({ nullable: false })
+	type!: RoomContentType;
+}
+
+@Entity({ tableName: 'roomcontents' })
+export class RoomContentEntity extends BaseEntityWithTimestamps {
```

Comment:
Why create a separate entity and not store with the room?

Reply:
AFAIK the room required to be independent from its content. I personally wouldn't be against it as long we are sure that we never need personalized lists etc. We also would have to take into account that we would have to store all other possible types of references in the room.

Reply:
Room content by `userId`.